Image classification [909] add image thumbnail + file name hover text + loading state

julia-rom opened the pull request:
- last part of the [909 card](https://trello.com/c/BHsApFYk/909-observation-table-2-8-thumbnail-and-file-name-display): add thumbnail to table
- added loading state to image drops
- update file structure in image classification folder: add image classification table and image upload modal subfolders
- image uploading will change once the api is ready, for now it is done in batch processing.

@alanjleonard I added a loading state in the modal - I propose keeping it here since it is an isolated process (modal is the dedicated area for file uploads - showing the progress here keeps things more visible/focused). It is also more likely to prevent the user from doing other interactions which might interrupt or conflict with the upload (clicking on other parts of the site/table). Just to clarify, image uploading should stay separate from the image processing.

In terms of processing, I think all image uploads should happen first because:
1) this gives a more predictable workflow (users can can fishing uploading before processing begins) 
2) we avoid complicated error handling (if something fails during upload or processing, it's easier to retry or manage without being mid-way through a combined operation) 
3) batch processing is faster

to test:
1. go to image classification table
2. upload new images (would recommend testing a large batch)
3. see loading state + modal close once finished
4. see thumbnails appear in column
5. hover over thumbnail to display file name text
6. click thumbnail to open up review modal

alanjleonard reviewed (changes requested):
We need to test the upload speed on slower connections if we're going to upload all of the image files _first_, then start processing them. The idea was that this will be used on slow connections, so the user could start an upload on 40 images, then start reviewing them as soon as the first image is done and the rest will continue to upload in the background. 

## Modal
- The user needs a way to cancel the upload process. Currently when you click the x, it closes the modal but still uploads everything. We can put a `Cancel upload` as a secondary button in the right of the modal. The x should do the same thing or remove it.
- Close button should be a secondary button on the right side to match other modals

## Table
- Thumbnails could be 150px square as a minimum. They all look the same at 50px. 
- Probably better if we create a 150px thumbnail, as opposed to using the full-sized image as the thumbnail.
- Clicking the thumbnail should also open the annotation modal (use `cursor: pointer`)

## Toast notification
- If you remove multiple images one right after the other, the toast notifications pile up and another one comes in as soon as you dismiss one. Maybe we can only show the last one if there are multiple of the exact same notifications?

julia-rom commented:
@alanjleonard 

modal
- for cancel upload, how should this button behave exactly? do we cancel all files (included ones that were already uploaded)? or do we leave existing files and cancel further uploads? what happens to mid-processed files if we go down that route?
- I removed the 'x' button from the image upload container

table
- updated to 150px x 150px
- added click handler to open up image annotation modal

toasts
- grouping them gets complicated (we have to track time between deletions, how many deletions in what time frame should count as a grouping?)
- we could display exact file name deleted for each file so it feels less redundant 

batch uploading/processing
- this might be good to further discuss with Kim. There are lots of things to consider. The big one is complex error handling - will require a backend/front end sync. How long does it take to process an image? Do we know this yet? What upload speeds should we test at? What happens if the connection fully cuts out mid way?

alanjleonard reviewed (changes requested):
- Cancel upload button should cancel the entire batch of images.
- There's no reason to have a disabled close button AND a cancel upload button. Omit the close button during the upload process. 
- Before uploading, the close button should be a secondary button on the right side to match other modals 
![image](https://github.com/user-attachments/assets/798062ce-49fb-4ae2-856b-9768c2888fb6)
- Add a 2px border to the images on hover. Use the callout colour.
-  Add a hover state to the drop area when dropping images. See orginal mockups below.
![image](https://github.com/user-attachments/assets/76c18670-010b-4d60-ba3b-d0a99eba01c7)

![image](https://github.com/user-attachments/assets/47a22083-91ff-4d55-9e6f-bda31b999e83)

julia-rom commented on src/components/pages/ImageClassification/ImageClassificationTable/Thumbnail.js:
```diff
@@ -0,0 +1,29 @@
+const Thumbnail = ({ imageUrl }) => {
+  return (
+    <ThumbnailWrapper>
+      <ThumbnailImage src={imageUrl} alt="Thumbnail" />
```
good call - will update

julia-rom commented:
@alanjleonard just added all the changes you mentioned
